pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -295,3 +275,40 @@ func existingVersionedResources(rs []ctlres.Resource) versionedResources {
 	}
 	return result
 }
+
+func newGroupedVersionedResources(rs []ctlres.Resource) map[string][]ctlres.Resource {
```

Comment:
I think I would prefer it if this function and the next one are either scoped to a struct, maybe we can move stuff that are used by other parts of the code to a single place at a later point. Not a blocker but a nice to have

Reply:
+1